src/mock_parsec/mod.rs

```diff
@@ -200,31 +184,37 @@ where
             .map(|(holder, _)| &***holder)
     }
 
-    fn update_blocks(&mut self) {
-        state::with(self.section_hash, |state| {
-            while let Some((block, holder)) = state.blocks.get(self.first_unconsensused) {
-                self.handle_consensus(block.payload());
+    fn compute_consensus(&mut self, state: &mut SectionState<T, S::PublicId>) {
+        loop {
+            state.compute_consensus(&self.peer_list, self.consensus_mode);
+
+            let mut recompute = false;
+
+            while let Some((block, holder)) = state.get_block(self.first_unconsensused) {
```

Comment:
Could probably be left in its own function `update_blocks`? to avoid nested loops.